src/app/pages/ArticlePage/PagePromoSections/TopStoriesSection/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useContext } from 'react';
+import { ServiceContext } from '#contexts/ServiceContext';
+import { arrayOf, shape } from 'prop-types';
+import { storyItem } from '#models/propTypes/storyItem';
+import pathOr from 'ramda/src/pathOr';
+import { C_GREY_2 } from '#app/legacy/psammead-styles/src/colours';
+import SectionLabel from '#legacy/psammead-section-label/src';
+import isEmpty from 'ramda/src/isEmpty';
+import {
+  StyledWrapper,
+  FlexPromoList,
+  FlexPromoListItem,
+} from './index.styles';
+import TopStoriesItem from './TopStoriesItem';
+
+const TopStoriesSection = ({ content }) => {
+  const { translations, script, service } = useContext(ServiceContext);
+
+  if (!content || isEmpty(content)) return null;
+
+  const title = pathOr('Top Stories', ['topStoriesTitle'], translations);
+  const hasSingleContent = content.length === 1;
+  const LABEL_ID = 'top-stories-heading';
+
+  return (
+    <StyledWrapper aria-labelledby={LABEL_ID} role="region" data-e2e={LABEL_ID}>
+      <SectionLabel
+        labelId={LABEL_ID}
+        columnType="secondary"
+        backgroundColor={C_GREY_2}
+        script={script}
+        service={service}
+      >
+        {title}
+      </SectionLabel>
+
+      {hasSingleContent ? (
+        <TopStoriesItem item={content[0]} index={0} />
+      ) : (
+        <FlexPromoList>
+          {content.map((item, index) => (
+            // eslint-disable-next-line react/no-array-index-key
```

Comment:
Yes, I believe if I construct the a11yId at this level, then I can use it. However, a11yId gets constructed with the index as well. I will read some docs about this. Never understood why it is bad to use index.

Reply:
Constructed it with the a11yId, unfortunately, the code is less readable because of this... Not sure what to do in this case. I also had a better reading on using index, and it should be ok when items are static and don't change position. In our case, the items will always take the same position. So it might be fine to use index. Maybe a question for everyone to answer